lib/manageiq/session/mem_cache_store_adapter.rb

```diff
@@ -0,0 +1,21 @@
+require "patches/rack_session_dalli_patch"
```

Comment:
Is it possible to delay load this now that we have a hook point?  That is, can we add some sort of initialize method to MemCacheStoreAdapter that then requires this patch?

Reply:
Same comment on the ActiveRecordStoreAdapter above

Reply:
Actually, maybe this isn't necessary, because with the safe_constantize change, these files aren't getting loaded until needed anyway.

Reply:
Yeah, I think that is the case.  I figured we actually got the best of both worlds with this since we are only loading what we need, and then prepending the module only when it is loaded.

Reply:
Really like the direction this is all going.
These patches and the session factory are getting really clean.

I still like the idea of extending these classes instead of patching them.

We would define `SessionActiveRecordStore` and use `session_store = :session_active_record_store` in config.
(or what ever name you decide)

Nick pointed out before that it is basically the same amount of code. So it seemed like a pedantic idea.
but now that the patches are so straight forward, I do want to put that out there again (sorry nick)